packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -146,5 +198,44 @@ String buildAotSnapshot(
   printStatus('Building snapshot...');
   runCheckedSync(genSnapshotCmd);
 
+  // On iOS, we use Xcode to compile the snapshot into a static library that the
+  // end-developer can link into their app.
+  if (platform == TargetPlatform.ios) {
+    printStatus('Building app.a...');
+
+    // These names are known to from the engine.
+    const String kDartVmIsolateSnapshotBuffer = 'kDartVmIsolateSnapshotBuffer';
+    const String kDartIsolateSnapshotBuffer = 'kDartIsolateSnapshotBuffer';
+
+    runCheckedSync(<String>['mv', vmIsolateSnapshot, path.join(outputDir.path, kDartVmIsolateSnapshotBuffer)]);
+    runCheckedSync(<String>['mv', isolateSnapshot, path.join(outputDir.path, kDartIsolateSnapshotBuffer)]);
+
+    String kDartVmIsolateSnapshotBufferC = path.join(outputDir.path, '$kDartVmIsolateSnapshotBuffer.c');
+    String kDartIsolateSnapshotBufferC = path.join(outputDir.path, '$kDartIsolateSnapshotBuffer.c');
+
+    runCheckedSync(<String>[
+      'xxd', '--include', kDartVmIsolateSnapshotBuffer, kDartVmIsolateSnapshotBufferC
+    ], workingDirectory: outputDir.path);
+    runCheckedSync(<String>[
+      'xxd', '--include', kDartIsolateSnapshotBuffer, kDartIsolateSnapshotBufferC
+    ], workingDirectory: outputDir.path);
+
+    String assemblyO = path.join(outputDir.path, 'snapshot_assembly.o');
+    String kDartVmIsolateSnapshotBufferO = path.join(outputDir.path, '$kDartVmIsolateSnapshotBuffer.o');
+    String kDartIsolateSnapshotBufferO = path.join(outputDir.path, '$kDartIsolateSnapshotBuffer.o');
+
+    runCheckedSync(<String>['xcrun', 'cc', assembly, '-o', assemblyO]);
```

Comment:
We can run these in `async` and await on all of them for completion before linking them in.